### PR TITLE
Make sure smoke tests aren't run on the rolling build

### PR DESCRIPTION
**Pull request by steveisok**
The way we were trying to use `isFullMatrix` was not visible to the xplat-setup template.  This PR makes sure we grab the value at run time as opposed to compile time.

### CODE REVIEW

**Comment by steveisok** [comment]
If you look at the expanded yml, `variables['isFullMatrix']` should be set.

**Comment by radical** [comment]
if we use a runtime expression(`$(isFullMatrix)`), then maybe we can sidestep the evaluation order issue?

**Comment by safern** [comment]
> if we use a runtime expression($(isFullMatrix)), then maybe we can sidestep the evaluation order issue?

Yes that's exactly what I suggested @steveisok offline and we would simplify the change.

File: eng/pipelines/common/xplat-setup.yml [hunk 1]
```diff
@@ -50,11 +50,7 @@ jobs:
         value: $(buildConfigUpper)
 
       - name: _runSmokeTestsOnlyArg
-        ${{ if ne(variables['nonPRBuild'], true) }}:
-          value: /p:RunSmokeTestsOnly=true
-        ${{ if eq(variables['nonPRBuild'], true) }}:
-          value: ''
-
+        value: '/p:RunSmokeTestsOnly=$(!isFullMatrix)'
```
Comment (safern) [inline eng/pipelines/common/xplat-setup.yml:53]: I don't think you can do: `!` here.

Comment (radical) [reply eng/pipelines/common/xplat-setup.yml:53]: would `$[ not($isFullMatrix) ]` work?

Comment (safern) [reply eng/pipelines/common/xplat-setup.yml:53]: That should work, but the syntax would be: `$[ not(variables.isFullMatrix) ]`

**Review by radical — APPROVED** [review]

**Pull request merged**